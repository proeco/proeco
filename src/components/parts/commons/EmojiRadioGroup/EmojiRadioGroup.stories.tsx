import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { EmojiRadioGroup } from './EmojiRadioGroup';

export default {
  title: 'parts/commons/EmojiRadioGroup',
  component: EmojiRadioGroup,
} as ComponentMeta<typeof EmojiRadioGroup>;

const Template: ComponentStory<typeof EmojiRadioGroup> = ({ selectedEmojiId, ...rest }) => {
  const [newSelectedEmojiId, setNewSelectedEmojiId] = useState(selectedEmojiId);

  const handleClickEmoji = (id: string) => {
    setNewSelectedEmojiId(id);
  };

  return (
    <div className="p-4">
      <EmojiRadioGroup selectedEmojiId={newSelectedEmojiId} {...rest} onClick={handleClickEmoji} />
    </div>
  );
};

export const Default = Template.bind({});
Default.args = {
  emojisInfo: [
    { emojiId: 'sob', count: 3 },
    { emojiId: 'confused', count: 10 },
    { emojiId: 'grinning', count: 6 },
    { emojiId: 'star-struck', count: 8 },
  ],
  selectedEmojiId: undefined,
};

export const ViewableEmojiRadioGroup = Template.bind({});
ViewableEmojiRadioGroup.args = {
  ...Default.args,
  viewable: true,
};
