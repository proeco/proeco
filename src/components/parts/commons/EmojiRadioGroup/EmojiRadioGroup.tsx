import React, { VFC } from 'react';
import styled from 'styled-components';

import { Emoji } from '~/components/parts/commons';

type Props = {
  emojisInfo: { emojiId: string; count: number }[];
  onClick: (id: string) => void;
  selectedEmojiId?: string;
  viewable?: boolean;
};

export const EmojiRadioGroup: VFC<Props> = ({ emojisInfo, onClick, selectedEmojiId, viewable = false }) => {
  return (
    <div className="d-flex align-items-center gap-3">
      {emojisInfo.map((emojiInfo, i) => {
        if (selectedEmojiId) {
          return (
            <div key={i} className="d-flex align-items-center">
              <SelectedEmojiBox selected={selectedEmojiId === emojiInfo.emojiId}>
                <Emoji emojiId={emojiInfo.emojiId} size={20} onClick={() => onClick(emojiInfo.emojiId)} />
              </SelectedEmojiBox>
              {viewable && <span className="ms-2 fs-2 fw-bold">{emojiInfo.count}</span>}
            </div>
          );
        }
        return (
          <div key={i} className="d-flex align-items-center">
            <StyledBox>
              <Emoji emojiId={emojiInfo.emojiId} size={20} onClick={() => onClick(emojiInfo.emojiId)} />
            </StyledBox>
            {viewable && <span className="ms-2 fs-2 fw-bold">{emojiInfo.count}</span>}
          </div>
        );
      })}
    </div>
  );
};

const StyledBox = styled.div`
  cursor: pointer;
  padding: 8px;
  transition: all ease 100ms;
  &:hover {
    transform: scale(1.3);
  }
`;

const SelectedEmojiBox = styled(StyledBox)<{ selected: boolean }>`
  filter: ${(props) => (props.selected ? 'grayscale(0%)' : 'grayscale(100%)')};
  -webkit-filter: ${(props) => (props.selected ? 'grayscale(0%)' : 'grayscale(100%)')};
  transform: ${(props) => (props.selected ? 'scale(1.35)' : 'scale(1)')};
  &:hover {
    filter: grayscale(0%);
    -webkit-filter: grayscale(0%);
  }
`;
