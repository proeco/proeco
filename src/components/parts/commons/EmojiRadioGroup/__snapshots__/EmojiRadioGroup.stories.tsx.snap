// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots parts/commons/EmojiRadioGroup Default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1ruxp1v"
      >
        <div
          class="MuiBox-root css-120nera"
        >
          <div
            class="MuiBox-root css-4rgiqd"
          >
            <div
              class="MuiBox-root css-186alq0"
            >
              <span
                aria-label="ðŸ˜­, sob"
                class="emoji-mart-emoji"
              >
                <span
                  style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 55.35714285714286% 41.07142857142858%;"
                />
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-51pf4d"
          >
            <div
              class="MuiBox-root css-186alq0"
            >
              <span
                aria-label="ðŸ˜•, confused"
                class="emoji-mart-emoji"
              >
                <span
                  style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 53.57142857142858% 100%;"
                />
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-51pf4d"
          >
            <div
              class="MuiBox-root css-186alq0"
            >
              <span
                aria-label="ðŸ˜€, grinning"
                class="emoji-mart-emoji"
              >
                <span
                  style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 53.57142857142858% 62.5%;"
                />
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-51pf4d"
          >
            <div
              class="MuiBox-root css-186alq0"
            >
              <span
                aria-label="ðŸ¤©, star-struck, grinning_face_with_star_eyes"
                class="emoji-mart-emoji"
              >
                <span
                  style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 67.85714285714286% 76.78571428571429%;"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1ruxp1v"
    >
      <div
        class="MuiBox-root css-120nera"
      >
        <div
          class="MuiBox-root css-4rgiqd"
        >
          <div
            class="MuiBox-root css-186alq0"
          >
            <span
              aria-label="ðŸ˜­, sob"
              class="emoji-mart-emoji"
            >
              <span
                style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 55.35714285714286% 41.07142857142858%;"
              />
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-51pf4d"
        >
          <div
            class="MuiBox-root css-186alq0"
          >
            <span
              aria-label="ðŸ˜•, confused"
              class="emoji-mart-emoji"
            >
              <span
                style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 53.57142857142858% 100%;"
              />
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-51pf4d"
        >
          <div
            class="MuiBox-root css-186alq0"
          >
            <span
              aria-label="ðŸ˜€, grinning"
              class="emoji-mart-emoji"
            >
              <span
                style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 53.57142857142858% 62.5%;"
              />
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-51pf4d"
        >
          <div
            class="MuiBox-root css-186alq0"
          >
            <span
              aria-label="ðŸ¤©, star-struck, grinning_face_with_star_eyes"
              class="emoji-mart-emoji"
            >
              <span
                style="width: 24px; height: 24px; display: inline-block; background-image: url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png); background-size: 5700% 5700%; background-position: 67.85714285714286% 76.78571428571429%;"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
